cmd/cloud.go

```diff
@@ -209,22 +219,45 @@ This will execute the test on the Load Impact cloud service. Use "k6 login cloud
 		signal.Notify(sigC, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
 		defer signal.Stop(sigC)
 
+		ticker := time.NewTicker(time.Millisecond * 2000)
+		shouldExitLoop := false
+
+		var (
+			startTime   time.Time
+			maxDuration time.Duration
+		)
+		if derivedConf.Duration.Valid {
+			maxDuration = time.Duration(derivedConf.Duration.Duration)
+		} else if derivedConf.Stages != nil {
+			maxDuration = lib.SumStagesDuration(derivedConf.Stages)
+		} else if derivedConf.Scenarios != nil {
+			et, _ := lib.NewExecutionTuple(nil, nil)
+			execReqs := derivedConf.Scenarios.GetFullExecutionRequirements(et)
+			maxDuration, _ = lib.GetEndOffset(execReqs)
+		}
```

Comment:
The thing about the `derived` configuration is that it's always going to include `scenarios`, that's its purpose. So, you can confidently simply use `GetFullExecutionRequirements()` - see https://github.com/loadimpact/k6/blob/41a4f730ad5db5dfbe399a6384351565460e1100/core/local/local.go#L65-L70

Also, `lib.NewExecutionTuple(nil, nil)` probably doesn't really matter, since the script duration isn't changed by changing the execution segment, but it's technically incorrect. Users are perfectly able to run `k6 cloud --execution-segment=30% script.js`, so this should be something like `lib.NewExecutionTuple(derivedConf.ExecutionSegment, derivedConf.ExecutionSegmentSequence)`, like: https://github.com/loadimpact/k6/blob/41a4f730ad5db5dfbe399a6384351565460e1100/core/local/local.go#L61

Reply:
OK, thanks for explaining. This should be resolved in 1c7f2ce5.